homeassistant/components/airthings_ble/coordinator.py

```diff
@@ -94,4 +99,35 @@ async def _async_update_data(self) -> AirthingsDevice:
             data = await self.airthings.update_device(self.ble_device)
         except Exception as err:
             raise UpdateFailed(f"Unable to fetch data: {err}") from err
+
+        await self._check_connectivity_mode_issue(data)
         return data
+
+    async def _check_connectivity_mode_issue(self, data: AirthingsDevice) -> None:
+        """Create or remove connectivity mode issue based on device data."""
+        if not (connectivity_mode := data.sensors.get("connectivity_mode")):
+            return
+
+        issue_id = f"smartlink_detected_{data.address}"
+
+        # Find sensors with connectivity mode set to smartlink (hub)
+        # or not configured
+        if connectivity_mode in [
+            AirthingsConnectivityMode.SMARTLINK.value,
+            AirthingsConnectivityMode.NOT_CONFIGURED.value,
+        ]:
```

Comment:
A set is faster for look ups.
```suggestion
        if connectivity_mode in {
            AirthingsConnectivityMode.SMARTLINK.value,
            AirthingsConnectivityMode.NOT_CONFIGURED.value,
        }:
```